tla/consensus/Traceccfraft.tla

```diff
@@ -254,6 +254,9 @@ IsRcvAppendEntriesResponse ==
                   \/ UpdateTerm(i, j, m) \cdot HandleAppendEntriesResponse(i, j, m)
                   \/ UpdateTerm(i, j, m) \cdot DropResponseWhenNotInState(i, j, m)
                   \/ DropResponseWhenNotInState(i, j, m)
+                  \* See comment on RcvAppendEntriesResponse in ccfraft
+                  \/ /\ m.success
+                     /\ DropStaleResponse(i, j, m)
```

Comment:
For posterity, this is the follow-on to the comment discussion here: https://github.com/microsoft/CCF/pull/5940#discussion_r1459142838

This special-case drop is now one level higher, duplicated in both the normal and trace validation specs, co-located with logically similar dropping decisions.